### PR TITLE
refactor(connector): changed amount to minor Unit for stripe

### PR DESCRIPTION
**Pull request by sahkal**
## Type of Change

- [x] Refactoring

## Description

This Pull request Uses the amount framework while sending request to stripe

Note: Stripe works in minor Unit

### Additional Changes

- [ ] This PR modifies the API contract
- [ ] This PR modifies the database schema
- [ ] This PR modifies application configuration/environment variables

## Motivation and Context

```shell
curl --location 'http://localhost:8080/payments' \
--header 'Content-Type: application/json' \
--header 'Accept: application/json' \
--header 'api-key: dev_OQjekPHEaK7qHI72VzXf63r6upc0sejXgi7GY5vOGsgwstsYtqMXvTv1bxwS5shF' \
--data-raw '
{
    "amount": 250,
    "currency": "USD",
    "confirm": true,
    "amount_to_capture": 250,
    "capture_method": "automatic",
    "capture_on": "2022-09-10T10:11:12Z",
    "customer_id": "StripeCustomer123",
    "email": "abcdef123@gmail.com",
    "name": "John Doe",
    "phone": "999999999",
    "phone_country_code": "+65",
    "description": "Please Pay for your XYZ",
    "authentication_type": "no_three_ds",
    "return_url": "https://google.com",
    "setup_future_usage": "off_session",
    "browser_info": {
        "user_agent": "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/70.0.3538.110 Safari/537.36",
        "accept_header": "text/html,application/xhtml+xml,application/xml;q=0.9,image/webp,image/apng,*/*;q=0.8",
        "language": "nl-NL",
        "color_depth": 24,
        "screen_height": 723,
        "screen_width": 1536,
        "time_zone": 0,
        "java_enabled": true,
        "java_script_enabled": true,
        "ip_address": "127.0.0.1"
    },
    "routing": {
        "type": "single",
        "data": "nmi"
    },
    "statement_descriptor_name": "joseph",
    "statement_descriptor_suffix": "JS",
    "metadata": {
        "udf1": "value1",
        "new_customer": "true",
        "login_date": "2019-09-10T10:11:12Z"
    },
    "payment_method": "card",
    "payment_method_type": "credit",
    "payment_method_data": {
        "card": {
            "card_number": "4242424242424242",
            "card_exp_month": "08",
            "card_exp_year": "25",
            "card_holder_name": "joseph Doe",
            "card_cvc": "999"
        }
    }
}'
```
Screenshot 2024-06-07 at 3 52 10 PM

## How did you test it?

Flows to test:

- Cards
- Bank Transfers
- ApplePay & Google Pay
- Refunds
- Payouts

## Checklist

- [x] I formatted the code `cargo +nightly fmt --all`
- [x] I addressed lints thrown by `cargo clippy`
- [x] I reviewed the submitted code
- [ ] I added unit tests for my changes where possible

### CODE REVIEW

File: crates/router/src/connector/stripe/transformers.rs [hunk 1]
```diff
@@ -2813,7 +2820,7 @@ pub struct StripeCashappQrResponse {
 
 #[derive(Clone, Debug, Eq, PartialEq, Deserialize, Serialize)]
 pub struct QrCodeResponse {
-    pub expires_at: Option<i64>,
+    pub expires_at: Option<MinorUnit>,
```
Comment (srujanchikke) [inline crates/router/src/connector/stripe/transformers.rs:2823]: This is supposed to be i64 not MinorUnit

File: crates/router/src/connector/utils.rs [hunk 2]
```diff
@@ -473,6 +474,12 @@ impl PaymentsPreProcessingData for types::PaymentsPreProcessingData {
     fn get_amount(&self) -> Result<i64, Error> {
         self.amount.ok_or_else(missing_field_err("amount"))
     }
+
+    // New minor amount function for amount framework
+    fn get_minor_amount(&self) -> Result<MinorUnit, Error> {
+        self.minor_amount.ok_or_else(missing_field_err("amount"))
+    }
```
Comment (srujanchikke) [inline crates/router/src/connector/utils.rs:481]: shouldn't it be 
```suggestion
        self.minor_amount.ok_or_else(missing_field_err("minor_amount"))
    }
```

File: crates/router/src/connector/stripe/transformers.rs [hunk 3]
```diff
@@ -2099,13 +2100,13 @@ pub struct SepaAndBacsBankTransferInstructions {
 #[derive(Clone, Debug, Serialize)]
 pub struct QrCodeNextInstructions {
     pub image_data_url: Url,
-    pub display_to_timestamp: Option<i64>,
+    pub display_to_timestamp: Option<MinorUnit>,
```
Comment (srujanchikke) [inline crates/router/src/connector/stripe/transformers.rs:2103]: display_to_timestamp needs to be in i64 not `MinorUnit`

**Review by srujanchikke — CHANGES_REQUESTED** [review]

File: crates/router/src/connector/utils.rs (hunk 2 above)
Comment (sahkal) [reply crates/router/src/connector/utils.rs:481]: no we shouldn't send this as a part of response

**Review by srujanchikke — APPROVED** [review]

**Review by srujanchikke — DISMISSED** [review]

**Review by AkshayaFoiger — COMMENTED** [review]
In the TryFrom for  charge request (line.no:3311), we are passing `value.request.amount`, please update it.

**Review by kashif-m — DISMISSED** [review]

**Review by AkshayaFoiger — APPROVED** [review]

**Review by Narayanbhat166 — APPROVED** [review]

**Review by Narayanbhat166 — DISMISSED** [review]

**Review by apoorvdixit88 — APPROVED** [review]
Dashboard specific changes are fine

**Review by jarnura — APPROVED** [review]

**Review by AkshayaFoiger — APPROVED** [review]

**Review by apoorvdixit88 — APPROVED** [review]

**Review by AkshayaFoiger — APPROVED** [review]

**Review by Narayanbhat166 — APPROVED** [review]

**Review by kashif-m — APPROVED** [review]

**Review by jarnura — APPROVED** [review]

**Pull request merged**